coral.py

```diff
@@ -90,9 +90,26 @@ def center_prompt(title, subtitle):
     if event.key == pygame.K_q:          # 'Q' quits game
         pygame.quit()
         sys.exit()
+##
+## Obstacle class
+##
+class Obstacle:
+    def __init__(self):
+        # Generate a random position for the obstacle within the game arena.
+        self.x = int(random.randint(0, WIDTH - GRID_SIZE) / GRID_SIZE) * GRID_SIZE
+        self.y = int(random.randint(0, HEIGHT - GRID_SIZE) / GRID_SIZE) * GRID_SIZE
+        # Create a rectangle object representing the obstacle.
+        self.rect = pygame.Rect(self.x, self.y, GRID_SIZE, GRID_SIZE)
+
+    def update(self):
+        # Draw the obstacle on the arena.
+        pygame.draw.rect(arena, "#FF4500", self.rect)  # Use orange color for obstacles.
```

Comment:
perhaps creating a constant for the obstacle color would fit better in the current code style, like we are already doing with the other objects colors